### PR TITLE
Inplace update events if possible for groupby query

### PR DESCRIPTION
**Pull request by navis**
we can avoid some map creations for group by queries.

### CODE REVIEW

File: processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategyV2.java [hunk 1]
```diff
@@ -115,14 +118,33 @@ public GroupByStrategyV2(
     final QueryGranularity gran = query.getGranularity();
     final String fudgeTimestamp;
     if (query.getContextValue(CTX_KEY_FUDGE_TIMESTAMP, "").isEmpty() && gran instanceof AllGranularity) {
-      final long timeStart = query.getIntervals().get(0).getStartMillis();
-      fudgeTimestamp = String.valueOf(
-          new DateTime(gran.iterable(timeStart, timeStart + 1).iterator().next()).getMillis()
-      );
+      fudgeTimestamp = String.valueOf(query.getIntervals().get(0).getStartMillis());
```
Comment (jaehc) [inline processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategyV2.java]: Does this have the same effect as the code removed?

Comment (navis) [inline processing/src/main/java/io/druid/query/groupby/strategy/GroupByStrategyV2.java]: I think so. for AllGranularity,
```
@Override
public Iterable<Long> iterable(long start, long end)
{
  return ImmutableList.of(start);
}
```

**Comment by gianm** [comment]
Hmm, are we breaking any contract of `Sequence` here? I originally did it with copying instead of in-place since I wasn't sure if it was okay to modify the return values of a sequence in-place.

If this is okay to do, seems like a good change since it should be faster.

**Comment by navis** [comment]
I don't think `Sequence` has any contract on internal datum mainly because it is for concealing it, imho. Hadoop has long history of reusing objects whenever possible and we might ride the tide.

**Pull request closed without merging**